### PR TITLE
docs: assigning pod to particular node

### PR DESCRIPTION
**Pull request by pdelewski**
##### Description

Fill in your description here.

---

##### Checklist

- [x] Changelog updated

###### Testing performed

- [x] Redeploy fluentd and fluentd-events pods
- [x] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/docs/Best_Practices.md [hunk 1]
```diff
@@ -1095,3 +1096,39 @@ The reliability of the system can be calculated using the following formulas:
...
+
+or by modyfing above settings in values.yaml
+
+```yaml
+fluent-bit:
+  nodeSelector:
+    kubernetes.io/os: linux
+```
+
+Node selectors for other pod types can be set in the same way.
```
Comment (pmalek-sumo) [inline deploy/docs/Best_Practices.md:1134]: Perhaps mention that using the keys as in the "table" above?

File: deploy/docs/Best_Practices.md [hunk 2]
```diff
@@ -1095,3 +1096,39 @@ The reliability of the system can be calculated using the following formulas:
...
+```
+  fluent-bit.nodeSelector.kubernetes.io/os=linux
+  fluentd.events.statefulset.nodeSelector.kubernetes.io/os=linux
+  fluentd.logs.statefulset.nodeSelector.kubernetes.io/os=linux
+  sumologic.setup.job.nodeSelector.kubernetes.io/os=linux
+  kube-prometheus-stack.prometheus-node-exporter.nodeSelector.kubernetes.io/os=linux
+  kube-state-metrics.nodeSelector.kubernetes.io/os=linux
+  prometheusOperator.nodeSelector.kubernetes.io/os=linux
+  prometheus.prometheusSpec.nodeSelector.kubernetes.io/os=linux
+```
```
Comment (pmalek-sumo) [inline deploy/docs/Best_Practices.md:1114]: Perhaps change this into a markdown table with name of the (sub)component like e.g. fluent-bit in the left column and the key for changing in `values.yaml` in the right?

File: deploy/docs/Best_Practices.md [hunk 3]
```diff
@@ -1095,3 +1096,39 @@ The reliability of the system can be calculated using the following formulas:
 
 - If limited by queue_size: `number_of_instances*timeout[min]*sending_queue.queue_size/load_in_DPM` minutes.
 - If limited by PVC size: `number_of_instances*PVC_size/(1KB*load_in_DPM)` minutes.
+
+## Assigning Pod to particular Node
```
Comment (pmalek-sumo) [inline deploy/docs/Best_Practices.md:1100]: I'd maybe split that into 2 levels: 1 for using `nodeSelector` and 1, lower level, for describing how to use it to pin pods to linux nodes. WDYT?

File: deploy/docs/Best_Practices.md (hunk 2 above)
Comment (sumo-drosiek) [reply deploy/docs/Best_Practices.md:1114]: I agree with @pmalek-sumo, eventually as a yaml style configuration with comments

File: deploy/docs/Best_Practices.md [hunk 4]
```diff
@@ -1095,3 +1098,55 @@ The reliability of the system can be calculated using the following formulas:
...
+Node selectors for other pod types can be set in the same way.
+
+| component                   | key                                                          | value |
+|-----------------------------|--------------------------------------------------------------|-------|
+| `otelagent`                 | `otelagent.daemonset.nodeSelector.kubernetes.io/os`          | linux |
+| `otelcol`                   | `otelcol.deployment.nodeSelector.kubernetes.io/os`           | linux |
+| `otelgateway`               | `otelgateway.deployment.nodeSelector.kubernetes.io/os`       | linux |
+| `otellogs`                  | `otellogs.daemonset.nodeSelector.kubernetes.io/os`           | linux |
+| `metadata`                  | `metadata.metrics.statefulset.nodeSelector.kubernetes.io/os` | linux |
+| `metadata`                  | `metadata.logs.statefulset.nodeSelector.kubernetes.io/os`    | linux |
```
Comment (sumo-drosiek) [inline deploy/docs/Best_Practices.md:1152]: Why are they two tables?

File: deploy/docs/Best_Practices.md [hunk 5]
```diff
@@ -1095,3 +1098,55 @@ The reliability of the system can be calculated using the following formulas:
...
+
+| component                   | key                                                                            | value |
+|-----------------------------|--------------------------------------------------------------------------------|-------|
+| `fluent-bit`                | `fluent-bit.nodeSelector.kubernetes.io/os`                                     | linux |
+| `fluentd`                   | `fluentd.events.statefulset.nodeSelector.kubernetes.io/os`                     | linux |
+| `fluentd`                   | `fluentd.logs.statefulset.nodeSelector.kubernetes.io/os`                       | linux |
+| `sumologic`                 | `sumologic.setup.job.nodeSelector.kubernetes.io/os`                            | linux |
+| `kube-prometheus-stack`     | `kube-prometheus-stack.prometheus-node-exporter.nodeSelector.kubernetes.io/os` | linux |
+| `kube-state-metrics`        | `kube-state-metrics.nodeSelector.kubernetes.io/os`                             | linux |
+| `prometheus`                | `prometheus.prometheusSpec.nodeSelector.kubernetes.io/os`                      | linux |
```
Comment (pmalek-sumo) [inline deploy/docs/Best_Practices.md:1123]: I like the table :)

One thing though: I'd drop the `linux` column and move it to the section above as a lookup table for people that want to find a key for a particular component.

File: deploy/docs/Best_Practices.md [hunk 6]
```diff
@@ -1095,3 +1098,55 @@ The reliability of the system can be calculated using the following formulas:
...
+### Binding pods to linux nodes
+
+Using this feature we can bind them to linux nodes.
+In order to do that `nodeSelector` has to be used. By default node selectors can be set for below pods:
+
+| component                   | key                                                                            | value |
+|-----------------------------|--------------------------------------------------------------------------------|-------|
+| `fluent-bit`                | `fluent-bit.nodeSelector.kubernetes.io/os`                                     | linux |
+| `fluentd`                   | `fluentd.events.statefulset.nodeSelector.kubernetes.io/os`                     | linux |
+| `fluentd`                   | `fluentd.logs.statefulset.nodeSelector.kubernetes.io/os`                       | linux |
```
Comment (pmalek-sumo) [inline deploy/docs/Best_Practices.md:1119]:
```suggestion
| `fluentd` (events)          | `fluentd.events.statefulset.nodeSelector.kubernetes.io/os`                     | linux |
| `fluentd` (logs)            | `fluentd.logs.statefulset.nodeSelector.kubernetes.io/os`                       | linux |
```

What about fluentd metrics?

File: deploy/docs/Best_Practices.md [hunk 7]
```diff
@@ -1095,3 +1098,55 @@ The reliability of the system can be calculated using the following formulas:
...
+
+## Assigning Pod to particular Node
+
+### Using NodeSelectors
+
+Kubernetes offers a feature of assigning specific pod to node. Such kind of control is sometimes useful,
+whenever you want to ensure that pod will end up on specific node according your requirements like operating system
+or connected devices.
+
+### Binding pods to linux nodes
```
Comment (pmalek-sumo) [inline deploy/docs/Best_Practices.md:1110]:
```suggestion
#### Binding pods to linux nodes
```

This IMHO should be nested under the section above.

**Review by pmalek-sumo — CHANGES_REQUESTED** [review]

**Comment by sumo-drosiek** [comment]
I think this should be added to `README.md`s table as well

File: deploy/docs/Best_Practices.md (hunk 4 above)
Comment (pdelewski) [reply deploy/docs/Best_Practices.md:1152]: The second table contains all other components that have nodeSelector label. I don't know if this table should be here or not. That's the question

File: deploy/docs/Best_Practices.md (hunk 7 above)
Comment (pdelewski) [reply deploy/docs/Best_Practices.md:1110]: ok

File: deploy/docs/Best_Practices.md (hunk 6 above)
Comment (pdelewski) [reply deploy/docs/Best_Practices.md:1119]: right

File: deploy/docs/Best_Practices.md (hunk 4 above)
Comment (sumo-drosiek) [reply deploy/docs/Best_Practices.md]: I would merge them into one table. I don't see a point in making two groups of pods

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**